lib/lambda_ethereum_consensus/p2p/incoming_request_handler.ex

```diff
@@ -117,8 +119,46 @@ defmodule LambdaEthereumConsensus.P2P.IncomingRequestHandler do
     end
   end
 
-  def handle_req("beacon_blocks_by_range/2/ssz_snappy", _message_id, _message),
-    do: :not_implemented
+  def handle_req("beacon_blocks_by_range/2/ssz_snappy", message_id, message) do
+    with <<24, snappy_blocks_by_range_request::binary>> <- message,
+         {:ok, ssz_blocks_by_range_request} <- Snappy.decompress(snappy_blocks_by_range_request),
+         {:ok, blocks_by_range_request} <-
+           Ssz.from_ssz(ssz_blocks_by_range_request, SszTypes.BeaconBlocksByRangeRequest) do
+      blocks_by_range_request
+      |> inspect(limit: :infinity)
+      |> then(&"[Received BlocksByRange Request] '#{&1}'")
+      |> Logger.debug()
+
+      ## TODO: there should be check that the `start_slot` is not older than the `oldest_slot_with_block`
+      %SszTypes.BeaconBlocksByRangeRequest{start_slot: start_slot, count: count} =
+        blocks_by_range_request
+
+      slot_coverage = start_slot + count
+
+      blocks =
+        start_slot..slot_coverage
+        |> Enum.reduce_while([], fn slot, current_blocks ->
+          blocks = case StateStore.get_state_by_slot(slot) do
+            {:ok, state} -> current_blocks ++ BlockStore.get_blocks(state.block_roots)
+            {:error, _} -> current_blocks
+            :not_found -> current_blocks
+          end
```

Comment:
This returns the latest block received in a slot.

Reply:
That's how it works now, but in the future, it will return the block corresponding to that slot.

Reply:
Yeah, I was a bit unclear how many blocks should be in a slot. Is this defined?

Reply:
0~1 blocks per slot, as per the spec. While fork choice is validating the blocks there might be many, but this is quickly resolved, since:
- either they are blocks proposed by a non-proposer (which are quickly discarded),
- or they are from the proposer and the network should choose only one on the next block (and the proposer will be slashed)

Reply:
Thank you. Please send a link where I can read up.

Reply:
The information is in https://eth2book.info/capella/part2/consensus/, but I don't know if there's a single page that explains this.

Reply:
@Godspower-Eze I just found out the `BeaconBlocksByRange` request doesn't send N chunks for N blocks:

https://github.com/lambdaclass/lambda_ethereum_consensus/blob/15617fd64f66daacf52b2e1545ad2a6efb81dd27/docs/specs/p2p-interface.md?plain=1#L898-L899

In this case, we should find a way to represent the "no blocks for this slot" in our DB, and parse that result accordingly here (not returning blocks for empty slots, instead of returning a response code of 3).

Reply:
Gotcha